### ISSUE TITLE
ubuntu 20.04 not connecting with windows-10 64 bit

### ISSUE THREAD

**Issue by ishika261** [debauchee/barrier#1188]
i wish to use ubuntu as my server and windows as a client. configuration and SSL keys matching is done. The client keeps shows barrier is running and Ubuntu(server) keeps showing barrier is starting. But I am not able to use the same mouse and keyboard for my client. Any help would be great! (p.s. I am new to ubuntu.)

**Comment by ranjeetnooli** [comment]
even i am having the same issue. i have ubuntu 20.04.2 LTS as server and windows as client. earlier it used to work but now it does not even connect. but when i use windows as server and ubuntu as client it does connect but cursor wont show up when i navigate to ubuntu machine from windows. earlier i have used barrier across my three systems, ubuntu as server and windows and kali linux as clients. never had a problem till now. :(

**Comment by presire** [comment]
I also have the same problem.
Server : openSUSE Leap 15.3 (x11)
Client : Windows 10

Like you, I can connect and it works fine with the following environment.
Server : Windows 10
Client : openSUSE Leap 15.3 (x11)

By the way, it worked fine in the previous environment.
Server : openSUSE Leap 15.2 (x11)
Client : Windows 10

**Comment by presire** [comment]
I don't know if this is the right way,
when I stopped firewalld, Barrier worked.

    Server : openSUSE Leap 15.3 (Plasma x11)
    Client : Windows 10

Perhaps, I think that this is a KVM issue.
On VMware, both are working fine.

![ksnip_20210626-090741](https://user-images.githubusercontent.com/82971634/123496729-83b9d600-d664-11eb-87ad-c80f69b20b29.png)
![無題](https://user-images.githubusercontent.com/82971634/123496731-84eb0300-d664-11eb-846f-b864398a91c1.png)

**Comment by ranjeetnooli** [comment]
Yes it does work when u stop firewall.. but after a while again it disconnects.. Everytime it disconnects I need to stop and restart the firewall.

**Comment by presire** [comment]
After a lot of trial and error,
By editing the following step, I was able to get Barrier to work.
<br>

First, create the ```/usr/lib/firewalld/services/barrier.xml``` file. (**You may not need this file**)
   
    Command :
    sudo nano /usr/lib/firewalld/services/barrier.xml
<br>

```
<?xml version="1.0" encoding="utf-8"?>
<service>
   <short>Synergy</short>
   <description>Share a mouse and keyboard on KVM.</description>
   <port protocol="tcp" port="24800"/>
</service>
```
<br>

Next, edit the ```/usr/lib/firewalld/zones/libvirt.xml``` file.
Above ```</zone>```, add ```<port port="24800" protocol="tcp"/>```.

    Command :
    sudo nano /usr/lib/firewalld/zones/libvirt.xml
<br>

```
<?xml version="1.0" encoding="utf-8"?>
<zone target="ACCEPT">
   ...
   ...
   <port port="24800" protocol="tcp"/>
</zone>
```

<br>

Finally, to reflect the firewall settings, run the following command.
Or reboot your PC.

    sudo firewall-cmd --reload
    sudo systemctl restart firewalld
<br>

There are some redundant parts,
by the above steps, worked for Barrier and Synergy.

**Comment by ranjeetnooli** [comment]
I don't know if this is ethical but this worked for me.

sudo firewall-cmd --permanent --add-port="24800/tcp"

after this it is working fine, automatically connects to both of my client machines ( windows and ubuntu )

**Comment by presire** [comment]
On ubuntu 20.04, Barrier was able to connect with <code>firewall-cmd</code> command.
I think that's the best (correct) way to do it.

On my openSUSE Leap 15.3, I could not connect with <code>firewall-cmd</code> command.

    sudo firewall-cmd --permanent --add-port="24800/tcp"
    sudo firewall-cmd --reload
<br>

Keep in mind that each distribution has a different method.

**Comment by ishika261** [comment]
> On ubuntu 20.04, Barrier was able to connect with `firewall-cmd` command.
> I think that's the best (correct) way to do it.
> 
> ```
> sudo firewall-cmd --permanent --add-port="24800/tcp"
> sudo firewall-cmd --reload
> ```
> 
> 
> On my openSUSE Leap 15.3, I could not connect with `firewall-cmd` command.
> (Maybe it is because I did not specify `--zone=libvirt`)
> 
> I would like to keep in mind that there are different methods for each distribution.

firewall-cmd is not working for me . i am getting this sudo: firewall-cmd: command not found

what to do?

**Comment by ranjeetnooli** [comment]
> > On ubuntu 20.04, Barrier was able to connect with `firewall-cmd` command.
> > I think that's the best (correct) way to do it.
> > ```
> > sudo firewall-cmd --permanent --add-port="24800/tcp"
> > sudo firewall-cmd --reload
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > On my openSUSE Leap 15.3, I could not connect with `firewall-cmd` command.
> > (Maybe it is because I did not specify `--zone=libvirt`)
> > I would like to keep in mind that there are different methods for each distribution.
> 
> firewall-cmd is not working for me . i am getting this sudo: firewall-cmd: command not found
> 
> what to do?

first install firewalld using,

$ sudo apt install firewalld

then start and enable firewalld

$ sudo systemctl start firewalld
$ sudo systemctl enable firewalld

then use this two commands

$ sudo firewall-cmd --permanent --add-port="24800/tcp"
$ sudo firewall-cmd --reload
